rails/app/controllers/external_activities_controller.rb

```diff
@@ -327,8 +327,71 @@ def copy
     end
   end
 
+  def edit_collections
+    authorize @external_activity
+
+    @collections = MaterialsCollection.includes(:materials_collection_items).order(:name).all
+
+    @material = [@external_activity]
+    @assigned_collections = @collections.select{|c| _collection_has_materials(c, @material) }
+
+    @unassigned_collections = @collections - @assigned_collections
+
+    respond_to do |format|
+      format.html # edit_collections.html.haml
+    end
+  end
+
+  def update_collections
+    authorize @external_activity
+
+    @external_activity = ExternalActivity.find(params[:material_id])
+    material_type = params[:material_type]
```

Comment:
It would be better to remove these lines. Just like in `edit_collections` the `@external_activity` should be setup automatically based on the id in URL.